src/stubs/sm_output.c

```diff
@@ -16,7 +16,7 @@ SM_FUNC(SM_NAME) void __sm_send_output(io_index index,
         continue;
 
       uint8_t* payload = malloc(payload_len);
-      if (payload == NULL)
+      if (payload == NULL  || !sancus_is_outside_sm(SM_NAME, (void *) payload, payload_len))
```

Comment:
we don't have to free the memory if the buffer is inside the SM, right?

Also, should we just raise an error if this happens (how?) instead of continuing?